### PR TITLE
fix: segment compatibility

### PR DESCRIPTION
**Pull request by Shahroz16**
Problem:
Our fork and Segment SDK can't be added at the same time via SPM.

Solution:
Update module name from `Segment` to `CioAnalytics`.

Changes:
- Package.json for updated module name
- Podspec for updated module name
- External Module reference of `Segment` 
- Sample Apps
- Objc symbols to avoid linker issues because of same objc names

### CODE REVIEW

**Comment by PaulWen** [comment]
Hi @Shahroz16, I just tried to upgrade from CIO 2x to CIO 3x but I am now blocked as we already have the Segment SDK in place and can't add this library at the same time via SPM.

Two questions:

1. What does CIO use this library for?
2. When is this PR expected to be merged or is there any alternative way to upgrade to CIO 3x while also having the Segment SDK installed via SPM?

Thanks!

**Comment by Shahroz16** [comment]
Hey @PaulWen, thank you for your feedback. As https://github.com/customerio/customerio-ios/issues/670, we are currently evaluating the effort required to support both SDKs running in parallel, but we haven't finalized a solution yet.

In the meantime, you can [continue using V2](https://github.com/customerio/customerio-ios/releases/tag/2.13.0), as it is fully supported and compatible with analytics-swift SDK. We will keep you updated on any developments. 

Since V3 [offers support for our CDP](https://customer.io/data-pipelines), we utilize features of Segment SDK. If you have any additional feedback or more use cases to share regarding the integration of analytics-swift SDK with V3 (as in our using both CIO CDP and Segment or just wanted to utilize some of v2 features of CIO), please do share with us. If it's easier to share privately you can email [product@customer.io](mailto:product@customer.io) to start a thread with our product team.

File: AnalyticsSwiftCIO.podspec [hunk 1]
```diff
@@ -7,14 +7,15 @@ Pod::Spec.new do |s|
     s.authors = "Customer.io"
 
     # Make sure the tag is the same as what SPM customers are using.
-    s.source = { :git => 'https://github.com/customerio/cdp-analytics-swift.git', :tag => '1.5.11+cio.1' }
+    # update branch to version
+    s.source = { :git => 'https://github.com/customerio/cdp-analytics-swift.git', :branch => 'shahroz/different-target' }
```
Comment (mrehan27) [inline AnalyticsSwiftCIO.podspec]: Adding comment to make sure we don't miss it.

**Review by mrehan27 — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**